bindings/rascal/utils/__init__.py

```diff
@@ -11,23 +7,19 @@
     dump_obj,
     load_obj,
 )
+from .misc import is_notebook
+
+# Warning potential dependency loop: FPS imports models, which imports KRR,
+# which imports this file again (including is_notebook() above)
+from .fps import fps, FPSFilter
+
+# TODO(max-veit) not needed...?
+# from ..lib import utils
+from ..lib._rascal.utils import ostream_redirect
+from copy import deepcopy
 from .cur import CURFilter
 from .scorer import get_score, print_score
 
 # function to redirect c++'s standard output to python's one
-ostream_redirect = utils.__dict__["ostream_redirect"]
-
-
-def is_notebook():
-    from IPython import get_ipython
-
-    try:
-        shell = get_ipython().__class__.__name__
-        if shell == "ZMQInteractiveShell":
-            return True  # Jupyter notebook or qtconsole
-        elif shell == "TerminalInteractiveShell":
-            return False  # Terminal running IPython
-        else:
-            return False  # Other type (?)
-    except NameError:
-        return False  # Probably standard Python interpreter
+# TODO(max-veit) we already have this imported...?
```

Comment:
same about TODO

Reply:
Ah, I had this because the following statement seemed to be a re-phrasing of `from rascal._rascal.utils import ostream_redirect` and is therefore not needed. Would you agree?

Reply:
that does look like it is doing the same thing yes. It is a bit weird that both are around. Maybe @felixmusil knows more about why this was here?

Reply:
Seems it's used almost exclusively in reference-data scripts, so it's not critical either way. I'll just take it out then.